client/lib/wpcom-undocumented/lib/site.js

```diff
@@ -183,6 +184,20 @@ UndocumentedSite.prototype.setOption = function( query, callback ) {
 	);
 }
 
+UndocumentedSite.prototype.postCounts = function( options, callback ) {
+	var type,
+		query = merge( {
+			type: 'post',
+			apiNamespace: 'wp', // This currently has no bearing on the route, wpcom-xhr-request needs to be updated to remove the namespace whitelist
+			locale: i18n.getLocaleSlug()
+		}, options );
+
+	type = query.type;
+	delete query.type;
+
+	this.wpcom.req.get( '/sites/' + this._id + '/wpcom/v2/post-counts/' + type, query, callback );
```

Comment:
Recommend `return`-ing this value, as then it can be used in a Promise chain:

`wpcom.undocumented().site().postCounts().then( () => { /* ... */ } );`